source/compute_engine/cuda/cuda_engine.cpp

```diff
@@ -37,6 +37,14 @@ CUDAEngine::CUDAEngine(const DeviceIndex deviceIndex, const uint32_t queueCount)
     Logger::logDebug("Initializing CUDA context");
     context = std::make_unique<CUDAContext>(devices.at(deviceIndex).getDevice());
 
+    // Set the GPU architecture for the CUDA device
+    int computeCapabilityMajor = 0;
+    int computeCapabilityMinor = 0;
+    cudaDeviceGetAttribute(&computeCapabilityMajor, cudaDevAttrComputeCapabilityMajor, deviceIndex);
+    cudaDeviceGetAttribute(&computeCapabilityMinor, cudaDevAttrComputeCapabilityMinor, deviceIndex);
+    std::string gpuArchitecture = std::string("--gpu-architecture=compute_") + std::to_string(computeCapabilityMajor) + std::to_string(computeCapabilityMinor);
```

Comment:
It would be better to replace cudaDeviceGetAttribute() with cuDeviceGetAttribute() here, so that we do not have runtime API dependency. E.g. cuDeviceGetAttribute(&computeCapabilityMajor, CU_DEVICE_ATTRIBUTE_COMPUTE_CAPABILITY_MAJOR, context->GetDevice()). The dependency can be also removed from premake script.

Reply:
Yep, that makes total sense, will fix it 👍

Reply:
It's changed now, but should also the functions be wrapped in `checkCUDAError`?

Reply:
Yeah, the wrapper method is generally used, although in this case it's not as critical.

Additionaly, I noticed that the last function argument is probably wrong - it should be reference to CUDA device instead of just index (e.g. context->getDevice() instead of deviceIndex).